app/conversations/routes.py

```diff
@@ -145,3 +144,23 @@ def get_conversations():
     response = {"conversations": results}
 
     return jsonify(response), 200
+
+
+@bp.route("/conversation/<conversation_uuid>", methods=["GET"])
+@cross_origin()
+def get_conversation(conversation_uuid):
+    """
+    Get a single conversation.
+    """
+    session_uuid = request.headers.get("X-Session-Id")
+    session_uuid = validate_uuid(session_uuid, uuidType.SESSION)
+    check_uuid_in_db(session_uuid, uuidType.SESSION)
+
+    validate_uuid(conversation_uuid, uuidType.CONVERSATION)
+    check_uuid_in_db(conversation_uuid, uuidType.CONVERSATION)
+    response = build_single_conversation_response(conversation_uuid)
+
+    start_user_b_journey(conversation_uuid)
```

Comment:
I think this is a smooth way to have user b events logged for analytics. But in this specific case I think this endpoint is used to retrieve the conversation card information to populate for user a? If so I don’t think this is a user B journey event and is a user A journey event. User A journey events are probably best to be tracked by the frontend since it’s already been tracking user A events with google analytics.

Also, when we want to add new analytic events, before it gets coded up we should add the new analytic event type and it’s description (whether frontend or backend) to the analytics page on Confluence so analytics people know what the event type means (they can read a description for each event type when it appears in the analytics data). This also helps to ensure we all understand and on same page for what a specific event is and when it should be triggered.

Reply:
This endpoint will be called when user b uses the unique link. It is unclear whether this will ever be part of user a's journey at all at the moment. But this is the structure agreed with frontend at the moment.

I was unaware of the existence of 'the analytics page' and do not want to be responsible for maintaining. The events I coded up in this round are the ones that you and I agreed, and the list is not exhaustive. Analytics do not need to be involved in any naming conventions in the backend. I can, however, change the name of the events going into the tables if you feel that they are not self-explanatory. I personally think they were clear enough, and changing them would be hair-splitting.

Reply:
Having double-checked, I can see that I misremembered the values for the effect and solutions choices though, so I will update those to UUIDs rather than the IRIs.